services/repository/delete.go

```diff
@@ -54,9 +54,11 @@ func DeleteRepositoryDirectly(ctx context.Context, doer *user_model.User, uid, r
 	}
 
 	// In case is a organization.
-	org, err := user_model.GetUserByID(ctx, uid)
-	if err != nil {
-		return err
+	var org *user_model.User
+	if uid != 0 {
```

Comment:
uid should not be zero. an error should be returned but not ignored.

Reply:
then we need some flag to make it able to be ignored ... I'll add one

Reply:
@lunny changed it